drivers/hmis/lib/hmis_data_cleanup/fix_enrollment_dates_20240416.rb

```diff
@@ -0,0 +1,91 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+class FixEnrollmentDates20240416
+  # For Enrollments in any project EXCEPT special treatment project, where the Entry Date and Exit Date are the same:
+  # - Add one day to the exit date
+  # For Enrollments in special treatment project, where the Exit Date is before the Entry Date (or they are the same):
+  # - If exit date is in 2020, change it to 2023 but same month and day.
+  # For Bed Night Services (recordType:200) in special treatment project, where the DateProvided is in 2020
+  # - Change DateProvided to 2023 (same month and day). The project was not open in 2020
+
+  include ArelHelper
+  attr_accessor :special_treatment_project_id
+  def initialize(special_treatment_project_id:)
+    self.special_treatment_project_id = special_treatment_project_id
+  end
+
+  def perform
+    puts "Fixing dates. Special treatment project: #{special_treatment_project_id}"
+    special_project = Hmis::Hud::Project.find_by(project_id: special_treatment_project_id)
```

Comment:
Missed one -- there could be projects in other data sources with the same `ProjectID`. The field, like all the HUD keys, are only guaranteed to be unique within the data source.
```suggestion
    special_project = Hmis::Hud::Project.hmis.find_by(project_id: special_treatment_project_id)
```

Reply:
@gigxz is this also needed for the Service query on line 54?

Reply:
oh yep, it is. It's needed because of using `ProjectID` to scope down the Services